rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -960,6 +961,58 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> guiSettings, std:
 
 	xemu_settings->EnhanceCheckBox(ui->spuDebug, emu_settings::SPUDebug);
 	SubscribeTooltip(ui->spuDebug, json_debug["spuDebug"].toString());
+		
+	if (utils::is_haswell_or_broadwell()!=utils::ModelNotHaswellBroadwell && utils::has_rtm())
+	{
+		xemu_settings->EnhanceCheckBox(ui->enableTSX, emu_settings::EnableTSX);
+		SubscribeTooltip(ui->enableTSX, json_debug["enableTSX"].toString());
+
+		// connect the toogled signal so that the stateChanged signal in EnhanceCheckBox can be prevented
+		connect(ui->enableTSX, &QCheckBox::toggled, [this](bool checked)
+		{
+			// create a messagebox depending on the CPU Model if the checkbox was checked
+			if (utils::is_haswell_or_broadwell() == utils::ModelHaswell)
+			{
+				if (checked && QMessageBox::No == QMessageBox::critical(this, tr("Haswell TSX Warning"), tr(
+					R"(
+					<p style="white-space: nowrap;">
+						RPCS3 has detected you are using TSX functions on a Haswell CPU.<br>
+						Intel has deactivated these functions in newer Microcode revisions, since they can lead to unpredicted behaviour.<br>
+						That means using TSX may break games or even <font color="red"><b>damage</b></font> your data.<br>
+						We recommend to disable this feature and update your computer BIOS.<br><br>
+						Do you wish to use TSX anyway?
```

Comment:
This line is different from the Broadwell version for no good reason

Reply:
Difference is: Haswell TSX is removed by Intel in new Microcode/BIOS versions. Enable TSX on Haswell is unsafe.
Broadwell TSX was fixed by Intel in new Microcode/BIOS versions. Enable TSX on Broadwell is safe.

Reply:
Referring to the very last line

Reply:
fixed it.